util/fibers/synchronization.h

```diff
@@ -404,16 +406,28 @@ class BlockingCounter {
       }
     }
 
+    auto WaitCondition(uint64_t* cnt) {
+      return [this, cnt]() -> bool {
+        *cnt = count_.load(std::memory_order_acquire);
+        return *cnt == 0 || (*cnt & kCancelFlag);
+      };
+    }
```

Comment:
It can't be declared below because the type can't then be inferred in Wait(), the alternative is implementing a custom struct with operator() below